smslibrary/src/main/java/com/eis0/smslibrary/SMSHandler.java

```diff
@@ -1,85 +1,98 @@
 package com.eis0.smslibrary;
 
-import android.annotation.TargetApi;
 import android.app.PendingIntent;
 import android.service.notification.NotificationListenerService;
 import android.service.notification.StatusBarNotification;
 import android.telephony.SmsMessage;
-import android.util.Log;
 
 import java.util.ArrayList;
 
-@TargetApi(21)
+/**
+ * Class that receives SMSes from the Core, analyzes them and calls the listeners.
+ * It also manage the message delivery system.
+ *
+ * @author Matteo Carnelos
+ * @author Marco Cognolato
+ */
 public class SMSHandler extends NotificationListenerService {
 
-    private static final char APP_ID = (char)0x02;
-    private static final String LOG_KEY = "SMS_HANDLER";
+    private static final char APP_ID = '\r';
     private static ArrayList<SMSMessage> pendingMessages = new ArrayList<>();
 
-    private static ReceivedMessageListener smsReceivedListener;
+    private static ReceivedMessageListener receivedListener;
 
     /**
-     * Sends a valid SMSmessage, with sent and delivery confirmation.
+     * Sends a valid SMSMessage, with sent and delivery confirmation.
+     *
      * @param message SMSMessage to send to the SMSPeer.
-     * @param sent PendingIntent listening for message sent
-     * @param delivered PendingIntent listening for message delivery
-     * @throws IllegalArgumentException if the destination is invalid
+     * @param sent PendingIntent listening for message sent.
+     * @param delivered PendingIntent listening for message delivery.
+     * @author Matteo Carnelos
+     * @author Marco Cognolato
      */
-    protected static void sendMessage(SMSMessage message, PendingIntent sent, PendingIntent delivered) {
+    static void sendMessage(SMSMessage message, PendingIntent sent, PendingIntent delivered) {
         SMSPeer destination = message.getPeer();
         SMSMessage msg = new SMSMessage(destination, APP_ID + message.getData());
-        if(!destination.isValid()) {
-            Log.e(LOG_KEY,"Invalid destination \"" + destination + "\"");
-            throw new IllegalArgumentException("Invalid destination \"" + destination + "\"");
-        }
```

Comment:
Hai rimosso il controllo di validità dall'interno di sendMessage, ma quindi dove lo hai piazzato?

Reply:
L'ho messo alla creazione di un SMSPeer (nel costruttore). In questo modo `SMSHandler.sendMessage(...)` riceve come argomento solo SMSPeer già validi (altrimenti non esisterebbero), e quindi non serve più fare controlli sul Peer.